### DISCUSSION TITLE
onTouched is not working

### DISCUSSION THREAD

**Discussion by MrFarhan** [react-hook-form/react-hook-form#10114] [Q&A]
### Version Number

^7.43.5

### Codesandbox/Expo snack

https://codesandbox.io/s/determined-haze-68ddx2?file=/login.js:275-284

### Steps to reproduce

`OnTouched` is not working, as per documentation:

> OnTouched Validation will trigger on the first blur event. After that, it will trigger on every change event.

Steps: type wrong email it will not show error, on blur it will show error, after user clicks again on the field and if user types proper email the error will be removed else error will show.

![image](https://user-images.githubusercontent.com/26528236/225377436-0257fe03-e85e-44af-8be9-9de5a76011ef.png)

### Expected behaviour

Error should first show on blur and then it works on onChange. 

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Expected behaviour.

**Comment by MrFarhan** [comment]
## Expected behaviour
#### Error should first show on blur and then it works on onChange.

Eg: user types `farhan@gmail` and click on other field
Error should show: `Invalid email`
when user again click on email field and remove all content then error should show `email is required` and if user types `farhan@gmail.com` error should be vanished.

Inshort on first it let the user type what he wants to type, but as he leave the input (i.e on blur) it should show the error that incorrect email. and When user revisit the email field and start typing the error should work on onChange i.e if he removes all text error should show `Email is required`, if he types incorrect email then the error should be `Incorrect email` , and as he types the proper email error should be vanished.

**Comment by bluebill1049** [comment]
Refer to this example: https://codesandbox.io/s/react-hook-form-get-started-forked-j13piq?file=/src/index.js
